### DISCUSSION TITLE
[Adonis 4.1] Error on view loop (Edge template)

### DISCUSSION THREAD

**Discussion by brandonsousa** [adonisjs/core#1432] [Help] [answered]
my view is looping for a result that is larger than expected and still with error and where there is no error it is duplicated, if I put it to be passed to the view with .toJSON it will not

**back**
![image](https://user-images.githubusercontent.com/34509230/89752516-1d0ee680-daab-11ea-9d43-2a108077e753.png)

**front**
![image](https://user-images.githubusercontent.com/34509230/89752687-c2c25580-daab-11ea-92e7-1afced2fa88b.png)

**view with json**
![view with json](https://user-images.githubusercontent.com/34509230/89752617-85f65e80-daab-11ea-9e63-17887ecb5412.PNG)

**view without json**
![view](https://user-images.githubusercontent.com/34509230/89752640-93134d80-daab-11ea-9ff7-4de63d29cde8.PNG)

**Comment by thetutlage** [comment]
I don't get whats the issue here?

You say result is larger than expected. what is expected?
Also what error do you get?

**Reply by brandonsousa** [reply]
the error is that i am only returning 1 bd value and in the view when i pass each it is showing more components on screen, and it is repeating and creating others that do not load the data, as in the print of the back and front withou json

**Comment by McSneaky** [comment, marked as answer]
TLDR: You are looping over object keys, not array

`findBy` returns single result, like you can see, it's just an object, not an array:
> Find a record using a key/value pair. Returns the first matching record.
https://adonisjs.com/docs/4.0/lucid#_findby_findbyorfail

Now when you pass it as `receipts:receipt` to view `receipts` is single object, not an array
Next you are looping over Object, which has 11 keys. 

Most likely you want to make it `await Receipt.query().where({user_id: user.id}).fetch()`

**Reply by brandonsousa** [reply]
thanks
